components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -965,52 +966,38 @@ subroutine radiation_init(state,pbuf)
-         deallocate(aerosol_optical_property)
-         allocate(aerosol_optical_property(pcols,pver,nswbands,begchunk:endchunk))
-         call get_aerosol_optical_property_from_file(next_month,'AER_SSA_SW','swband',nswbands,aerosol_optical_property)
-         call pbuf_set_field(pbuf,aer_ssa_bnd_sw_mon_2_idx,aerosol_optical_property)
-         deallocate(aerosol_optical_property)
-
-         allocate(aerosol_optical_property(pcols,pver,nswbands,begchunk:endchunk))
-         call get_aerosol_optical_property_from_file(current_month,'AER_G_SW','swband',nswbands,aerosol_optical_property)
-         call pbuf_set_field(pbuf,aer_asm_bnd_sw_mon_1_idx,aerosol_optical_property)
-         deallocate(aerosol_optical_property)
-         allocate(aerosol_optical_property(pcols,pver,nswbands,begchunk:endchunk))
-         call get_aerosol_optical_property_from_file(next_month,'AER_G_SW','swband',nswbands,aerosol_optical_property)
-         call pbuf_set_field(pbuf,aer_asm_bnd_sw_mon_2_idx,aerosol_optical_property)
-         deallocate(aerosol_optical_property)
+         allocate(aerosol_optical_property_lw(pcols,pver,nlwbands,begchunk:endchunk))
+         
+         call get_aerosol_optical_property_from_file(current_month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property_lw)
+         call pbuf_set_field(pbuf, aer_tau_bnd_lw_mon_1_idx,aerosol_optical_property_lw)
+         call get_aerosol_optical_property_from_file(next_month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property_lw)
+         call pbuf_set_field(pbuf,aer_tau_bnd_lw_mon_2_idx,aerosol_optical_property_lw)
+
+         allocate(aerosol_optical_property_sw(pcols,pver,nswbands,begchunk:endchunk))
+         
+         call get_aerosol_optical_property_from_file(current_month,'AER_TAU_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf, aer_tau_bnd_sw_mon_1_idx,aerosol_optical_property_sw)
+         call get_aerosol_optical_property_from_file(next_month,'AER_TAU_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf,aer_tau_bnd_sw_mon_2_idx,aerosol_optical_property_sw)
+
+         call get_aerosol_optical_property_from_file(current_month,'AER_SSA_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf, aer_ssa_bnd_sw_mon_1_idx,aerosol_optical_property_sw)
+         call get_aerosol_optical_property_from_file(next_month,'AER_SSA_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf,aer_ssa_bnd_sw_mon_2_idx,aerosol_optical_property_sw)
+
+         call get_aerosol_optical_property_from_file(current_month,'AER_G_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf,aer_asm_bnd_sw_mon_1_idx,aerosol_optical_property_sw)
+         call get_aerosol_optical_property_from_file(next_month,'AER_G_SW','swband',nswbands,aerosol_optical_property_sw)
+         call pbuf_set_field(pbuf,aer_asm_bnd_sw_mon_2_idx,aerosol_optical_property_sw)
```

Comment:
Do you need deallocate statements here?

Reply:
it worked without deallocation but I added it just in case.

Reply:
I think this kind of stuff causes memory leaks. Maybe not in this case since this is just in init, but probably in the radiation_tend case where you allocate every month but never deallocate. In any case, fixed now.